server/linter/analyzer/analyzer_entities.go

```diff
@@ -40,18 +58,130 @@ func (l Linter) Validate() error {
 	}
 
 	for _, p := range l.Plugins {
-		if p.Name == "" {
-			return fmt.Errorf("plugin name cannot be empty")
+		plugin, ok := findPlugin(p.Id, DefaultPlugins)
+		if !ok {
+			return fmt.Errorf("plugin %s not supported, supported plugins are %s", p.Id, strings.Join(AvailablePlugins, " | "))
+		}
+
+		for _, r := range p.Rules {
+			index := slices.IndexFunc(plugin.Rules, func(rule LinterRule) bool { return rule.Id == r.Id })
+
+			if index == -1 {
+				availableRules := strings.Join(getAvailableRules(plugin), " | ")
+				return fmt.Errorf("rule %s not found for plugin %s, supported rules for plugin are %s", r.Id, p.Id, availableRules)
+			}
 		}
 	}
 
 	return nil
 }
 
+func findPlugin(Id string, plugins []LinterPlugin) (LinterPlugin, bool) {
+	for _, p := range plugins {
+		if p.Id == Id {
+			return p, true
+		}
+	}
+
+	return LinterPlugin{}, false
+}
+
+func findRule(Id string, rules []LinterRule) (LinterRule, bool) {
+	for _, r := range rules {
+		if r.Id == Id {
+			return r, true
+		}
+	}
+
+	return LinterRule{}, false
+}
+
+func getAvailableRules(plugin LinterPlugin) []string {
+	rules := make([]string, 0)
+
+	for _, r := range plugin.Rules {
+		rules = append(rules, r.Id)
+	}
+
+	return rules
+}
+
+func getDefaultLinter() Linter {
+	return Linter{
+		ID:           id.ID("current"),
+		Name:         "analyzer",
+		Enabled:      true,
+		Plugins:      DefaultPlugins,
+		MinimumScore: 0,
+	}
+}
+
 func (l Linter) HasID() bool {
 	return l.ID != ""
 }
 
 func (l Linter) GetID() id.ID {
 	return l.ID
 }
+
+func (l Linter) EnabledPlugins() []LinterPlugin {
+	plugins := make([]LinterPlugin, 0)
+	for _, p := range l.Plugins {
+		if p.Enabled {
+			plugins = append(plugins, p)
+		}
+	}
+
+	return plugins
+}
+
+func (l Linter) ShouldSkip() bool {
+	return !l.Enabled
+}
+
+func (l Linter) WithMetadata() (Linter, error) {
```

Comment:
Combines the analyzer configuration (Database, CLI, FE) with the metadata stored within the repo, as it is not available for users to update (today)